lib/res_controller.js

```diff
@@ -0,0 +1,47 @@
+class ResController {
+    constructor() {
+        this.userResponses = {};
+    }
+
+    addUserRequest(params) {
+        params = Object.assign({}, params);
+
+        if (this.userResponses[params.ip] === undefined) this.userResponses[params.ip] = [];
+
+        this.userResponses[params.ip].push(params);
+    }
+
+    respondToUser(ip, params, res_response, status_code) {
```

Comment:
I was confused by the iteration this function was performing at first. Here's a (clearer IMO) alternative:
```js
respondToUser(ip, params, res_response, status_code) {
    if (this.userResponses[ip] === undefined) return;

    // Find the request in our array
    let responseIndex = this.userResponses[ip].findIndex((item) => {
        for (let param in params) {
            if (params[param] != item[param]) return false;
        }

        return true;
    })

    let response = this.userResponses[ip][responseIndex]

    // Send back the data
    if (params.type == "http") {
        if (!status_code) response["res"].json(res_response);
        else response["res"].status(status_code).json(res_response);
    }
    else if (params.type == "ws") {
        response["res"].emit('floatmessage', res_response);
    }

    // Remove the request
    this.userResponses[ip].splice(responseIndex, 1);
}
```